term/src/terminalstate/mod.rs

```diff
@@ -665,6 +678,26 @@ impl TerminalState {
             .unwrap_or_else(|| self.config.color_palette())
     }
 
+    /// Returns a copy of the current mouse cursor shape.
+    pub fn get_mouse_cursor_shape(&self) -> MouseCursor {
+        // FIXME: The log here is clearing a cache somewhere causing the mouse shape
+        //        to be updated correctly. Not sure what I need to change.
+        log::info!("Mouse cursor shape: {}", self.mouse_cursor_shape);
```

Comment:
I noticed this sort of by accident. I was trying to figure out why the cursor shape wouldn't change after closing an app that supports OSC 22 codes:

https://github.com/user-attachments/assets/8cb5310b-6bb5-462a-9fe4-c57283dc787d

I wanted to see if the cursor shape wasn't being updated, but it turns out it is! And logging the value clears some cache I'm not aware of that causes the cursor to be drawn correctly up the call stack.

I don't know what I'm missing, so I'd appreciate some direction on how this could be addressed 🙏